Sources/Moya/MoyaProvider+Stubbing.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+
+/// Mark: Stubbing
+
+/// A TargetType configurable for testing responses
+public protocol TestTargetType {
+
+    var stubBehavior: Moya.StubBehavior { get }
+
+    var sampleResponse: EndpointSampleResponse { get }
+
+}
+
+/// Controls how stub responses are returned.
+public enum StubBehavior {
+
+    /// Do not stub.
+    case never
+
+    /// Return a response immediately.
+    case immediate
+
+    /// Return a response after a delay.
+    case delayed(seconds: TimeInterval)
+}
```

Comment:
Really great ideas here ! Super nice work @SD10 .

Reply:
Thanks for the review!